backend/src/main/java/com/redhat/cloud/notifications/recipients/rbac/RbacRecipientUsersProvider.java

```diff
@@ -0,0 +1,79 @@
+package com.redhat.cloud.notifications.recipients.rbac;
+
+import com.redhat.cloud.notifications.recipients.User;
+import com.redhat.cloud.notifications.routers.models.Page;
+import io.quarkus.cache.CacheResult;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class RbacRecipientUsersProvider {
+
+    @Inject
+    @RestClient
+    RbacServiceToService rbacServiceToService;
+
+    @ConfigProperty(name = "recipient-provider.rbac.elements-per-page", defaultValue = "40")
+    Integer rbacElementsPerPage;
+
+    @CacheResult(cacheName = "rbac-recipient-users-provider-get-users")
+    public Uni<List<User>> getUsers(String accountId, boolean adminsOnly) {
+        return getWithPagination(
+                page -> rbacServiceToService
+                        .getUsers(accountId, adminsOnly, page * rbacElementsPerPage, rbacElementsPerPage)
+        ).memoize().indefinitely();
```

Comment:
Can you add a TODO saying that `.memoize().indefinitely()` should be removed after the Quarkus 2.0 bump?